clients/client-py/taskcluster/helper.py

```diff
@@ -0,0 +1,128 @@
+# -*- coding: utf-8 -*-
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import os
+
+import logging
+import json
+from taskcluster.generated import _client_importer
+
+logger = logging.getLogger(__name__)
+
+
+class TaskclusterConfig(object):
+    """
+    Local configuration used to access Taskcluster service and objects
+    """
+
+    def __init__(self, url="https://community-tc.services.mozilla.com"):
+        self.options = None
+        self.secrets = None
+        self.default_url = os.environ.get("TASKCLUSTER_ROOT_URL", url)
+
+    def auth(self, client_id=None, access_token=None, max_retries=12):
+        """
+        Build Taskcluster credentials options
+        Supports, by order of preference:
+         * directly provided credentials
+         * credentials from local configuration
+         * credentials from environment variables
+         * taskclusterProxy
+         * no authentication
+        """
+        self.options = {"maxRetries": max_retries}
+
+        if client_id is None and access_token is None:
+            # Credentials preference: Use local config from release-services
+            xdg = os.path.expanduser(os.environ.get("XDG_CONFIG_HOME", "~/.config"))
```

Comment:
I'm wary of this -- putting creds in cleartext in ~ suggests that they are long-lived credentials (more than a few days) and that they are fairly powerful (not tailored to one specific operation).  That's kind of dangerous, especially for the Firefox-CI deployment, and not a behavior we'd like to encourage -- that's largely why we removed support for such config files from the clients.  Would you be comfortable removing it here, too?

Reply:
Works for me, we are rarely using this feature anymore and if needs arise, we can still implement it in the user application